kuma/wiki/forms.py

```diff
@@ -295,6 +305,34 @@ def clean_slug(self):
 
         return slug
 
+    def clean_tags(self):
+        """
+        Validate the tags ensuring we have no case-sensitive duplicates.
+        """
+        tags = self.cleaned_data['tags']
+        dupe_tags = []
+
+        if tags:
+            for tag in parse_tags(tags):
+                # Note: The exact match query doesn't work correctly with
+                # MySQL with regards to case-sensitivity. If we move to
+                # Postgresql in the future this code may need to change.
+                doc_tag = DocumentTag.objects.filter(
+                    name__exact=tag).values_list('name', flat=True)
+                # Write a log we can grep to help find duplicate document tags.
+                if len(doc_tag) > 1:
```

Comment:
Sure. This is for the case of MySQL/Django finding more than one tag. E.g. if `tag='Javascript'` and `doc_tag == ['JavaScript', 'Javascript']`. I thought it would be helpful to be able to grep the logs like `grep 'duplicate document tags' <logfile> | uniq` to help us track down which ones we can remove so they don't get added back.